cyhy/db/database.py

```diff
@@ -1288,9 +1292,23 @@ def save(self, *args, **kwargs):
             self["agency"]["location"]["gnis_id"] = long(
                 self["agency"]["location"]["gnis_id"]
             )
+
         # Set enrollment date if it was not previously set
         if not self.get("enrolled"):
             self["enrolled"] = util.utcnow()
+
+        # Verify that all email address fields in fact
+        # contain valid email addresses.
+        contacts = self["agency"].get("contacts")
+        if contacts is not None:
+            for contact in contacts:
+                email_address = contact.get("email")
+                if email_address is not None:
+                    # This should be modified to use fullmatch() when
+                    # this code is ported to Python 3.
+                    if not self.__EmailAddressRegex.match(email_address):
+                        raise ValueError(email_address + " is not a valid email address")
```

Comment:
Since we're not doing anything special with the default return of `None` I think this would be cleaner.

```suggestion
        contacts = self["agency"].get("contacts", [])
        for contact in contacts:
            email_address = contact.get("email")
            if email_address is not None:
                # This should be modified to use fullmatch() when this code is
                # ported to Python 3.
                if not self.__EmailAddressRegex.match(email_address):
                    raise ValueError(email_address + " is not a valid email address")
```